File: apps/api/test/fixtures/multi-lang/Payment.java
// @ts-nocheck
package com.example.payment;

import java.util.logging.Logger;

public interface PaymentProcessor {
    boolean processPayment(double amount, String currency);
}

public class PaymentService implements PaymentProcessor {
    private static final Logger logger = Logger.getLogger(PaymentService.class.getName());
    private final String apiKey;
    private final int timeout;

    public PaymentService(String apiKey, int timeout) {
        this.apiKey = apiKey;
        this.timeout = timeout;
    }

    @Override
    public boolean processPayment(double amount, String currency) {
        logger.info("Processing " + amount + " " + currency);
        PaymentGateway gateway = new PaymentGateway(apiKey);
        return gateway.charge(amount, currency);
    }

    public void refund(String transactionId) {
        logger.info("Refunding " + transactionId);
    }
}

class PremiumPaymentService extends PaymentService {
    public PremiumPaymentService(String apiKey) {
        super(apiKey, 60);
    }

    public double applyDiscount(double amount, double discount) {
        return amount * (1 - discount);
    }
}

File: apps/api/test/fixtures/multi-lang/payment.ts
// @ts-nocheck
import { Logger } from './utils';
import { PaymentGateway } from './gateway';

export interface PaymentConfig {
  apiKey: string;
  timeout: number;
}

export class PaymentService {
  private logger: Logger;

  constructor(private config: PaymentConfig) {
    this.logger = new Logger('PaymentService');
  }

  async processPayment(amount: number, currency: string): Promise<boolean> {
    this.logger.info(`Processing ${amount} ${currency}`);
    const gateway = new PaymentGateway(this.config.apiKey);
    return gateway.charge(amount, currency);
  }

  refund(transactionId: string): void {
    this.logger.info(`Refunding ${transactionId}`);
  }
}

export function calculateTax(amount: number, rate: number): number {
  return amount * rate;
}
